scrape/scrape.go

```diff
@@ -326,24 +327,23 @@ func (sp *scrapePool) reload(cfg *config.ScrapeConfig) error {
 	targetScrapePoolReloads.Inc()
 	start := time.Now()
 
+	var wg sync.WaitGroup
+	sp.reloadMtx.Lock()
```

Comment:
What exactly is this lock protecting?

Reply:
Yes, I put it here but I need to check if we can call reload multiple times in the code. It is protecting having multiple reloads in parallel, which could be problematic if they have a long tail to shut down (many connections closing).